src/lib/actions/sendEmail.ts

```diff
@@ -0,0 +1,47 @@
+'use server';
+
+import EmailTemplate from '@/components/email-template';
+import { Resend } from 'resend';
+import { SendEmailStatuses, TSendEmailResponse, TSendEmailState } from './sendEmail.types';
+
+export async function sendEmail(prevState: TSendEmailState, formData: FormData): Promise<TSendEmailResponse> {
+  const resendKey = process.env.RESEND_API_KEY;
```

Comment:
[Major] Unhandled error, missing check if Resend api key exists, results in error.
![image](https://github.com/user-attachments/assets/c531023a-d5c7-4d01-8d2e-5f6366f6204a)

![image](https://github.com/user-attachments/assets/273f6864-6d6c-47a6-9120-e527fa9c0fa1)

Also let's remove `pages_legacy` directory, add a test that would mock external Resend API behaviour, check if error like this one is properly handeled by the Server Action

Reply:
I'm just wondering if there is any case when the API key could be `undefined`, since it came from `.env`. What do you think?

Reply:
It might for example expire